### ISSUE TITLE
Removing duplicates fails with [JavaScript Application] Failed erasing messages

### ISSUE THREAD

**Issue by komradebob** [eyalroz/removedupes#58]
Trying to erase dupes in a subfolder breaks as soon as I hit the 'ok' button.

Big Sur 11.4, tBird 78.12.0.

Any clues?

Thanks!

Error log contents:

[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
Successfully loaded OpenPGP library librnp.dylib version 0.14+git20210121.7c8492b4.MZLA from /Applications/Thunderbird.app/Contents/MacOS/librnp.dylib RNPLib.jsm:95:15
Found 0 public keys and 0 secret keys (0 protected, 0 unprotected) RNPLib.jsm:291:15
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
OpenGL compositor Initialized Succesfully.
Version: 2.1 INTEL-16.4.5
Vendor: Intel Inc.
Renderer: Intel(R) Iris(TM) Plus Graphics 640
FBO Texture Target: TEXTURE_2D
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
OpenGL compositor Initialized Succesfully.
Version: 2.1 INTEL-16.4.5
Vendor: Intel Inc.
Renderer: Intel(R) Iris(TM) Plus Graphics 640
FBO Texture Target: TEXTURE_2D
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
Trying to load /Applications/Thunderbird.app/Contents/MacOS/libotr.5.dylib OTRLib.jsm:51:11
Successfully loaded OTR library /Applications/Thunderbird.app/Contents/MacOS/libotr.5.dylib OTRLib.jsm:59:13
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
Error: Invalid autocomplete selectedIndex AutoCompleteChild.jsm:208:13
OpenGL compositor Initialized Succesfully.
Version: 2.1 INTEL-16.4.5
Vendor: Intel Inc.
Renderer: Intel(R) Iris(TM) Plus Graphics 640
FBO Texture Target: TEXTURE_2D
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.js:200
Uncaught DOMException: CustomElementRegistry.define: 'conversation-browser' has already been defined as a custom element conversation-browser.js:853
OpenGL compositor Initialized Succesfully.
Version: 2.1 INTEL-16.4.5
Vendor: Intel Inc.
Renderer: Intel(R) Iris(TM) Plus Graphics 640
FBO Texture Target: TEXTURE_2D
Prompter: Browser not available or tab modal prompts disabled. Falling back to window prompt. Prompter.jsm:1051
OpenGL compositor Initialized Succesfully.
Version: 2.1 INTEL-16.4.5
Vendor: Intel Inc.
Renderer: Intel(R) Iris(TM) Plus Graphics 640
FBO Texture Target: TEXTURE_2D
Prompter: Browser not available or tab modal prompts disabled. Falling back to window prompt. Prompter.jsm:1051
OpenGL compositor Initialized Succesfully.
Version: 2.1 INTEL-16.4.5
Vendor: Intel Inc.
Renderer: Intel(R) Iris(TM) Plus Graphics 640
FBO Texture Target: TEXTURE_2D
Uncaught 
Exception { name: "NS_ERROR_FAILURE", message: "Failure arg 0 [nsIMsgFolder.deleteMessages]", result: 2147500037, filename: "chrome://removedupes/content/removedupes-common.js", lineNumber: 464, columnNumber: 0, data: null, stack: "removeDupesFromSingleFolder@chrome://removedupes/content/removedupes-common.js:464:22\nremoveDuplicates@chrome://removedupes/content/removedupes-common.js:420:29\nonAccept@chrome://removedupes/content/removedupes-dialog.js:611:36\ninitDupeReviewDialog/<@chrome://removedupes/content/removedupes-dialog.js:130:63\n_fireButtonEvent@chrome://global/content/elements/dialog.js:487:19\n_doButtonCommand@chrome://global/content/elements/dialog.js:466:29\n_handleButtonCommand@chrome://global/content/elements/dialog.js:460:19\n", location: XPCWrappedNative_NoHelper }

**Comment by eyalroz** [comment]
Which version of the extension are you using?

**Comment by komradebob** [comment]
0.5.2b2

**Comment by eyalroz** [comment]
Ah, then - please try the build here:

https://github.com/eyalroz/removedupes/issues/57

**Comment by komradebob** [comment]
That one won't load. Drop it in but never get menu entry. 

error log: 
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128
Extension.jsm:570:20
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128
Extension.jsm:570:20
1627509772157	addons.xpi	WARN	Exception running bootstrap method update on {a300a000-5e21-4ee0-a115-9ec8f4eaa92b}: Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)(resource://gre/modules/Extension.jsm:570:20) JS Stack trace: readJSON/</<@Extension.jsm:570:20
onStopRequest@NetUtil.jsm:128:18
Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE) Extension.jsm:570
1627509772164	addons.xpi	WARN	Exception running bootstrap method startup on {a300a000-5e21-4ee0-a115-9ec8f4eaa92b}: Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)(resource://gre/modules/Extension.jsm:570:20) JS Stack trace: readJSON/</<@Extension.jsm:570:20
onStopRequest@NetUtil.jsm:128:18
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128
Extension.jsm:570:20
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128
Extension.jsm:570:20
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128

**Comment by eyalroz** [comment]
Can you try removing the extension, installing it again, then restatring TB? Those are pretty weird errors.

Also, when listing errors - before you do what you're trying to do, clear the error log, so we don't get the errors due to other code... :-(

**Comment by komradebob** [comment]
Uninstalled, restarted, cleared log, re-installed. Same result. Log follows:

Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE) Extension.jsm:570
1627510030755	addons.xpi	WARN	Exception running bootstrap method startup on {a300a000-5e21-4ee0-a115-9ec8f4eaa92b}: Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)(resource://gre/modules/Extension.jsm:570:20) JS Stack trace: readJSON/</<@Extension.jsm:570:20
onStopRequest@NetUtil.jsm:128:18
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128
Extension.jsm:570:20
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128
Extension.jsm:570:20
Uncaught (in promise) Error: Error while loading 'jar:file:///Users/bownes/Library/Thunderbird/Profiles/xom7mkp1.default/extensions/%7Ba300a000-5e21-4ee0-a115-9ec8f4eaa92b%7D.xpi/!/manifest.json' (NS_ERROR_FAILURE)
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128

**Comment by eyalroz** [comment]
Hmm. I can't even tell where the failure is occurring :-(

Anyway, and just to explain: The fix in 0.5.2b3 is that I had made this change to the array of headers for removal/move, to be TB-91 compatible, but did not ensure it would still work with TB 78. So, I added some conditionals ("if version is at least 79 do X else do Y"). On my system, this works both on TB 78 and TB 91. I can't really say why you're getting these weird errors - and when reading JSON even.

**Comment by komradebob** [comment]
Upgraded to Big Sur 11.5 and reverted to the original version and it works. The beta has the same behaviour. 

Many many thanks for your prompt efforts!

**Comment by eyalroz** [comment]
"same behavior" as in you see the error, or as in working like the original version?

**Comment by komradebob** [comment]
Same behavior as in doesn’t even show up in the add ins list, gets an error Didn’t check to see if it is the same. I can check in the morning. 

Thanks!

**Comment by eyalroz** [comment]
Let's try something else. Can you create a new TB profile? And if you can, does the error still manifest when you have only my extension installed on the new profile? 

Also, what other extensions are you using?

**Comment by davidrnewman** [comment]
I am getting the same error messages when I try to remove duplicates in local folders.

When I had it set to move deleted messages to the Deleted folder, I get the JavaScript message: 

> Failed moving messages to: 
> mailbox://nobody@Local%20Folders/Trash

When I set it to Delete permanently, I get the message:

> Failed erasing messages.

I am running Thunderbird 78.11.0 (64-bit) under Kubuntu 21.04. 

Remove Duplicate Messages is version 0.5.2b2

**Comment by eyalroz** [comment]
@davidrnewman : What you're seeing is a different issue  : #57 . That issue has been resolved in v0.5.2b3, which you can get from the issue page (and soon via addons.thunderbird.net)

**Comment by buerkelr** [comment]
Hi, 
I have the same problems:
RDM does not work since version 0.5.2b2.
Version 0.5.2b3 does also NOT work.

In some folders of my inbox RDM does not show any message.
I am getting this errors in the debug console:

```
Uncaught 
Exception { name: "NS_ERROR_FAILURE", message: "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIStringBundle.GetStringFromName]", result: 2147500037, filename: "chrome://removedupes/content/removedupes-common.js", lineNumber: 135, columnNumber: 0, data: null, stack: "getByName@chrome://removedupes/content/removedupes-common.js:135:24\nRemoveDupes.namedAlert@chrome://removedupes/content/removedupes-common.js:119:34\ntraverseSearchFolderSubfolders@chrome://removedupes/content/removedupes.js:232:21\naddSearchFolders@chrome://removedupes/content/removedupes.js:215:34\nbeginSearchForDuplicateMessages@chrome://removedupes/content/removedupes.js:115:36\nsearchAndRemoveDuplicateMessages@chrome://removedupes/content/removedupes.js:79:34\noncommand@chrome://messenger/content/messenger.xhtml:1:30\n", location: XPCWrappedNative_NoHelper }
removedupes-common.js:135



Uncaught 
Exception
​
columnNumber: 0
​
data: null
​
filename: "chrome://removedupes/content/removedupes-common.js"
​
lineNumber: 135
​
location: XPCWrappedNative_NoHelper
​​
QueryInterface: function QueryInterface()
​​
asyncCaller: null
​​
asyncCause: null
​​
caller: XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), filename: , name: , … }
​​
columnNumber: 24
​​
filename: "chrome://removedupes/content/removedupes-common.js"
​​
formattedStack: "getByName@chrome://removedupes/content/removedupes-common.js:135:24\nRemoveDupes.namedAlert@chrome://removedupes/content/removedupes-common.js:119:34\ntraverseSearchFolderSubfolders@chrome://removedupes/content/removedupes.js:232:21\naddSearchFolders@chrome://removedupes/content/removedupes.js:215:34\nbeginSearchForDuplicateMessages@chrome://removedupes/content/removedupes.js:115:36\nsearchAndRemoveDuplicateMessages@chrome://removedupes/content/removedupes.js:79:34\noncommand@chrome://messenger/content/messenger.xhtml:1:30\n"
​​
lineNumber: 135
​​
name: "getByName"
​​
nativeSavedFrame: SavedFrame { source: "chrome://removedupes/content/removedupes-common.js", sourceId: 571, line: 135, … }
​​
sourceId: 571
​​
sourceLine: ""
​​
toString: function toString()
​​
<get asyncCaller()>: function asyncCaller()
​​
<get asyncCause()>: function asyncCause()
​​
<get caller()>: function caller()
​​
<get columnNumber()>: function columnNumber()
​​
<get filename()>: function filename()
​​
<get formattedStack()>: function formattedStack()
​​
<get lineNumber()>: function lineNumber()
​​
<get name()>: function name()
​​
<get nativeSavedFrame()>: function nativeSavedFrame()
​​
<get sourceId()>: function sourceId()
​​
<get sourceLine()>: function sourceLine()
​​
<prototype>: Object { … }
​
message: "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIStringBundle.GetStringFromName]"
​
name: "NS_ERROR_FAILURE"
​
result: 2147500037
​
stack: "getByName@chrome://removedupes/content/removedupes-common.js:135:24\nRemoveDupes.namedAlert@chrome://removedupes/content/removedupes-common.js:119:34\ntraverseSearchFolderSubfolders@chrome://removedupes/content/removedupes.js:232:21\naddSearchFolders@chrome://removedupes/content/removedupes.js:215:34\nbeginSearchForDuplicateMessages@chrome://removedupes/content/removedupes.js:115:36\nsearchAndRemoveDuplicateMessages@chrome://removedupes/content/removedupes.js:79:34\noncommand@chrome://messenger/content/messenger.xhtml:1:30\n"
​
<prototype>: ExceptionPrototype { toString: toString(), name: Getter, message: Getter, … }
removedupes-common.js:135
    getByName chrome://removedupes/content/removedupes-common.js:135
    namedAlert chrome://removedupes/content/removedupes-common.js:119
    traverseSearchFolderSubfolders chrome://removedupes/content/removedupes.js:232
    addSearchFolders chrome://removedupes/content/removedupes.js:215
    beginSearchForDuplicateMessages chrome://removedupes/content/removedupes.js:115
    searchAndRemoveDuplicateMessages chrome://removedupes/content/removedupes.js:79
    oncommand chrome://messenger/content/messenger.xhtml:1

```

**Comment by buerkelr** [comment]
It seems the the upper error only happens in inbox folder with subfolders.
I have a local inbox folder with 11 subfolders.
RDM does not work in this inbox folder.
But RDM works in folder without subfolder!

**Comment by eyalroz** [comment]
Please try this new version:

https://github.com/eyalroz/removedupes/releases/tag/v0.5.2b4

**Comment by buerkelr** [comment]
Hi eyalroz,
the issue with the subfolders is fixed with version 0.5.2b4.
Thank you!
:-)

**Comment by eyalroz** [comment]
@buerkelr : Great, thanks for reporting this.
